cmd/kubeadm/app/apis/kubeadm/types.go

```diff
@@ -425,6 +425,9 @@ type ComponentConfig interface {
 	// DeepCopy should create a new deep copy of the component config in place
 	DeepCopy() ComponentConfig
 
+	// Version returns a Kubernetes API version of the component config (v1alpha1, v1beta1, v1, etc.)
+	Version() string
```

Comment:
i think we might want to have VersionKind() here instead even if our third-party CCs don't have multiple kinds yet.

Reply:
I don't want kinds to be visible at all (let alone at this level). What I was thinking, and I'll probably change it on an iteration is to have this as a `GroupVersion`.
The idea behind `ComponentConfig` objects is to have them hide the details around a given `GroupVersion`. This includes what kinds there are, what goes where and how to structure it.
This helps isolate version specific code. Plus, adding new component config versions becomes more easy.

Reply:
Groups are not supposed to change, but a Group can introduce new Kinds during upgrade to a new Version. not tracking new Kinds means no visibility for this aspect from "plan", correct?

GroupVersion-only tracking would mean that "plan" can say, hey i know there is a new version  for this Group, but i'm not going to tell you what changed in it.

Reply:
after discussions on this, having `group | version` is fine.